share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md

```diff
@@ -35,3 +35,16 @@ To make changes based on feedback:
 Once all feedback has been addressed and if it's a change we want to include (we include most changes) then we'll add your commit to Homebrew. Note it will not show up as "Merged" because of the way we include contributions.
 
 Well done, you are now a Homebrew contributor!
+
+Tips:
+
+* `pushd $(brew --prefix)` before doing any of the following; `popd` afterward.
+
+* If `brew update` doesn't seem to work, you left your branch checked out. If you're still working on it, `git rebase origin/master` after each `brew update`. To restore normality, `git checkout master`.
+
+* If `brew update` outputs tips on how to "restore the stashed changes", you had edited some files but not committed the changes. Follow the `git stash pop` advice to recover.
```

Comment:
I wonder if it's worth not recommending `brew update` above but instead recommending `git pull --rebase` (or `git fetch` and `git rebase`) directly to avoid this?